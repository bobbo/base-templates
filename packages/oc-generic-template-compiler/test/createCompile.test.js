const createCompile = require('../lib/createCompile');
const compileStatics = jest.fn();
const compileView = jest.fn();
const compileServer = jest.fn();
const getInfo = jest.fn();

test('Should correctly create a compiler function', () => {
  const compiler = createCompile({
    compileServer,
    compileView,
    compileStatics,
    getInfo
  });

  expect(compiler).toBeInstanceOf(Function);
  expect(compiler.length).toBe(2);
  expect(compiler.toString()).toMatchSnapshot();
});
