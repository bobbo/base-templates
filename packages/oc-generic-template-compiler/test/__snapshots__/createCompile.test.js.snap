// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly create a compiler function 1`] = `
"(
  options,
  callback) =>
  {/* istanbul ignore next */cov_1bx4i1wq1q.f[1]++;
    const componentPackage = /* istanbul ignore next */(cov_1bx4i1wq1q.s[7]++, _.cloneDeep(options.componentPackage));
    const ocPackage = /* istanbul ignore next */(cov_1bx4i1wq1q.s[8]++, options.ocPackage); /* istanbul ignore next */cov_1bx4i1wq1q.s[9]++;

    async.waterfall(
    [
    // Compile view
    function (cb) {/* istanbul ignore next */cov_1bx4i1wq1q.f[2]++;cov_1bx4i1wq1q.s[10]++;
      // compiledViewInfo, bundleInfo
      compileView(options, (error, compiledViewInfo) => {/* istanbul ignore next */cov_1bx4i1wq1q.f[3]++;cov_1bx4i1wq1q.s[11]++;
        if (error) {/* istanbul ignore next */cov_1bx4i1wq1q.b[0][0]++;cov_1bx4i1wq1q.s[12]++;
          return cb(error);
        } else /* istanbul ignore next */{cov_1bx4i1wq1q.b[0][1]++;}
        // USE COMPILATION INFO TO MASSAGE COMPONENT'S PACKAGE
        cov_1bx4i1wq1q.s[13]++;componentPackage.oc.files.template = compiledViewInfo.template; /* istanbul ignore next */cov_1bx4i1wq1q.s[14]++;
        delete componentPackage.oc.files.client; /* istanbul ignore next */cov_1bx4i1wq1q.s[15]++;
        cb(error, { componentPackage, compiledViewInfo });
      });
    },
    // Compile dataProvider
    function ({ componentPackage, compiledViewInfo }, cb) {/* istanbul ignore next */cov_1bx4i1wq1q.f[4]++;cov_1bx4i1wq1q.s[16]++;
      if (!componentPackage.oc.files.data) {/* istanbul ignore next */cov_1bx4i1wq1q.b[1][0]++;cov_1bx4i1wq1q.s[17]++;
        return cb(null, componentPackage);
      } else /* istanbul ignore next */{cov_1bx4i1wq1q.b[1][1]++;}cov_1bx4i1wq1q.s[18]++;
      compileServer(
      Object.assign({}, options, { compiledViewInfo }),
      (error, compiledServerInfo) => {/* istanbul ignore next */cov_1bx4i1wq1q.f[5]++;cov_1bx4i1wq1q.s[19]++;
        if (error) {/* istanbul ignore next */cov_1bx4i1wq1q.b[2][0]++;cov_1bx4i1wq1q.s[20]++;
          return cb(error);
        } else /* istanbul ignore next */{cov_1bx4i1wq1q.b[2][1]++;}
        // USE COMPILATION INFO TO MASSAGE COMPONENT'S PACKAGE
        cov_1bx4i1wq1q.s[21]++;componentPackage.oc.files.dataProvider = compiledServerInfo; /* istanbul ignore next */cov_1bx4i1wq1q.s[22]++;
        delete componentPackage.oc.files.data; /* istanbul ignore next */cov_1bx4i1wq1q.s[23]++;
        cb(error, componentPackage);
      });

    },
    // Compile package.json
    function (componentPackage, cb) {/* istanbul ignore next */cov_1bx4i1wq1q.f[6]++;cov_1bx4i1wq1q.s[24]++;
      componentPackage.oc.files.template.version = getInfo().version; /* istanbul ignore next */cov_1bx4i1wq1q.s[25]++;
      componentPackage.oc.version = ocPackage.version; /* istanbul ignore next */cov_1bx4i1wq1q.s[26]++;
      componentPackage.oc.packaged = true; /* istanbul ignore next */cov_1bx4i1wq1q.s[27]++;
      componentPackage.oc.date = getUnixUtcTimestamp(); /* istanbul ignore next */cov_1bx4i1wq1q.s[28]++;
      if (!componentPackage.oc.files.static) {/* istanbul ignore next */cov_1bx4i1wq1q.b[3][0]++;cov_1bx4i1wq1q.s[29]++;
        componentPackage.oc.files.static = [];
      } else /* istanbul ignore next */{cov_1bx4i1wq1q.b[3][1]++;}cov_1bx4i1wq1q.s[30]++;
      if (!_.isArray(componentPackage.oc.files.static)) {/* istanbul ignore next */cov_1bx4i1wq1q.b[4][0]++;cov_1bx4i1wq1q.s[31]++;
        componentPackage.oc.files.static = [componentPackage.oc.files.static];
      } else /* istanbul ignore next */{cov_1bx4i1wq1q.b[4][1]++;}cov_1bx4i1wq1q.s[32]++;
      fs.writeJson(
      path.join(options.publishPath, 'package.json'),
      componentPackage,
      error => {/* istanbul ignore next */cov_1bx4i1wq1q.f[7]++;cov_1bx4i1wq1q.s[33]++;
        cb(error, componentPackage);
      });

    },
    // Compile statics
    function (componentPackage, cb) {/* istanbul ignore next */cov_1bx4i1wq1q.f[8]++;cov_1bx4i1wq1q.s[34]++;
      compileStatics(options, error => {/* istanbul ignore next */cov_1bx4i1wq1q.f[9]++;cov_1bx4i1wq1q.s[35]++;return cb(error, componentPackage);});
    }],

    callback);

  }"
`;
